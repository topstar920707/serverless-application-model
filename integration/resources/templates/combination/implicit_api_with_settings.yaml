Globals:
  Api:
    EndpointConfiguration: REGIONAL
    BinaryMediaTypes:
      - image~1jpg
      - image~1png
    MethodSettings: [{
        "LoggingLevel": "INFO",
        "MetricsEnabled": True,
        "DataTraceEnabled": True,
        "ResourcePath": "/*",
        "HttpMethod": "*"
    }]

Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs12.x
      CodeUri: ${codeuri}
      MemorySize: 128

      Events:
        GetApi:
          Type: Api
          Properties:
            Path: /pathget
            Method: get
