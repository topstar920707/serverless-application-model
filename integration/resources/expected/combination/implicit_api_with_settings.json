[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyLambdaFunctionGetApiPermissionProd", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"ServerlessRestApiDeployment", "ResourceType":"AWS::ApiGateway::Deployment" },
  { "LogicalResourceId":"ServerlessRestApi", "ResourceType":"AWS::ApiGateway::RestApi" },
  { "LogicalResourceId":"ServerlessRestApiProdStage", "ResourceType":"AWS::ApiGateway::Stage" }
]